### PR TITLE
DEV: Support for running theme test with Ember CLI

### PR DESCRIPTION
**Pull request by eviltrout**
This is quite complex as it means that in production we have to build
Ember CLI test files and allow them to be used by our Rails application.

There is a fair bit of glue we can remove in the future once we move to
Ember CLI completely.

### CODE REVIEW

File: app/helpers/qunit_helper.rb [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+    if @legacy_ember
+      result << preload_script("discourse/tests/test_starter")
+    else
+      if @is_proxied
+        result << preload_script("scripts/discourse-test-listen-boot")
+        result << preload_script("scripts/discourse-boot")
+      else
+        result << preload_script("discourse-test-listen-boot")
+        result << preload_script("discourse-boot")
+      end
```
Comment (CvX) [inline app/helpers/qunit_helper.rb:28]: So far so good! Just a minor readability suggestion:

```suggestion
    elsif @is_proxied
      result << preload_script("scripts/discourse-test-listen-boot")
      result << preload_script("scripts/discourse-boot")
    else
      result << preload_script("discourse-test-listen-boot")
      result << preload_script("discourse-boot")
```

**Comment by eviltrout** [comment]
(looks like I have a couple tests to fix next week!)

**Review by CvX — COMMENTED** [review]
`public/images/emoji/google/smiling_face_with_three_hearts.png` got into the PR

**Review by davidtaylorhq — COMMENTED** [review]
This looks great to me! I tried it out locally, and in a production sandbox, and everything seems to work as expected.

The increase in build time is unfortunate, but we're doing a lot more work, so it makes sense.

I tried updating `ember-auto-import`, and it seems like cold-cache builds are a *little* faster, and warm-cache builds are a **lot** faster, so that might help to counteract the increase.

To upgrade, I ran:

```
yarn add "webpack@5"
yarn upgrade "ember-auto-import@^2.2.4"
```

**Comment by eviltrout** [comment]
@davidtaylorhq almost all the increase in build time is the result of `forbidEval`. I have a follow up TODO to investigate that.

**Comment by eviltrout** [comment]
@davidtaylorhq @CvX no approvals? 😢

**Comment by eviltrout** [comment]
@CvX wow I missed that *and* your comment. I will fix.

**Review by CvX — APPROVED** [review]

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**